tests/sentry/utils/test_flag.py

```diff
@@ -0,0 +1,129 @@
+import asyncio
+import concurrent.futures as cf
+
+from django.urls import reverse
+
+from sentry.testutils.cases import APITestCase
+from sentry.utils import flag
+
+
+def test_flag_tracking():
+    """Assert the ring buffer works."""
+    flag.initialize_flag_manager(capacity=3)
+
+    flag.process_flag_result("a", True)
+    flags = flag.get_flags_serialized()
+    assert len(flags) == 1
+    assert flags == [{"flag": "a", "result": True}]
+
+    flag.process_flag_result("b", True)
+    flags = flag.get_flags_serialized()
+    assert len(flags) == 2
+    assert flags == [{"flag": "a", "result": True}, {"flag": "b", "result": True}]
+
+    flag.process_flag_result("c", True)
+    flags = flag.get_flags_serialized()
+    assert len(flags) == 3
+    assert flags == [
+        {"flag": "a", "result": True},
+        {"flag": "b", "result": True},
+        {"flag": "c", "result": True},
+    ]
+
+    flag.process_flag_result("d", False)
+    flags = flag.get_flags_serialized()
+    assert len(flags) == 3
+    assert flags == [
+        {"flag": "b", "result": True},
+        {"flag": "c", "result": True},
+        {"flag": "d", "result": False},
+    ]
+
+    flag.process_flag_result("e", False)
+    flag.process_flag_result("f", False)
+    flags = flag.get_flags_serialized()
+    assert len(flags) == 3
+    assert flags == [
+        {"flag": "d", "result": False},
+        {"flag": "e", "result": False},
+        {"flag": "f", "result": False},
+    ]
```

Comment:
actually @ryan953 check out this test case: it looks like the behavior is as intended